src/components/Tooltip/Tooltip.tsx

```diff
@@ -105,32 +106,39 @@ export const Tooltip = ({
     const timeoutRef = useRef<ReturnType<typeof setTimeout> | null>(null);
 
     useEffect(() => {
-        const newOffset = getNewOffsetBasedOnArrowPosition(currentPlacement, offset);
+        const newOffset = withArrow ? getNewOffsetBasedOnArrowPosition(currentPlacement, offset) : offset;
 
-        if (newOffset[0] !== tooltipOffset[0] || newOffset[1] !== tooltipOffset[1]) {
-            setTooltipOffset(newOffset);
-        }
-    }, [offset, tooltipOffset, currentPlacement]);
+        setTooltipOffset((prevOffset) => {
+            if (newOffset[0] !== prevOffset[0] || newOffset[1] !== prevOffset[1]) {
+                return newOffset;
+            }
+            return prevOffset;
+        });
+    }, [currentPlacement, offset, withArrow]);
 
     const handleHideTooltip = useCallback(() => {
-        if (timeoutRef.current) {
-            clearTimeout(timeoutRef.current);
+        if (!disabled) {
+            if (timeoutRef.current) {
```

Comment:
Could you make a reusable function for this? and replace this clear and set timeout